net/HttpRequest.cpp

```diff
@@ -553,6 +553,12 @@ int64_t Response::readData(const char* p, int64_t len)
                     if (digit < 0)
                         break;
 
+                    // Can assume that digit is always less than 16.
+                    if (chunkLen >= std::numeric_limits<int64_t>::max() / 16)
+                    {
+                        // Would not fit into chunkLen.
+                        return len - available;
```

Comment:
Wouldn't this be a fatal error? Would be be able to recover? If not, best to return -1 and fail.

Reply:
Good point, -1 is better here, this is really a garbage input, we just should not crash on this.